lib/ROM/RomParser.cpp

```diff
@@ -0,0 +1,95 @@
+// ****************************************************************************** //
+//   _______   __                              __       __                        //
+//  /       \ /  |                            /  \     /  |                       //
+//  $$$$$$$  |$$ |  ______   __    __         $$  \   /$$ |  ______   _______     //
+//  $$ |__$$ |$$ | /      \ /  |  /  | ______ $$$  \ /$$$ | /      \ /       \    //
+//  $$    $$/ $$ | $$$$$$  |$$ |  $$ |/      |$$$$  /$$$$ | $$$$$$  |$$$$$$$  |   //
+//  $$$$$$$/  $$ | /    $$ |$$ |  $$ |$$$$$$/ $$ $$ $$/$$ | /    $$ |$$ |  $$ |   //
+//  $$ |      $$ |/$$$$$$$ |$$ \__$$ |        $$ |$$$/ $$ |/$$$$$$$ |$$ |  $$ |   //
+//  $$ |      $$ |$$    $$ |$$    $$ |        $$ | $/  $$ |$$    $$ |$$ |  $$ |   //
+//  $$/       $$/  $$$$$$$/  $$$$$$$ |        $$/      $$/  $$$$$$$/ $$/   $$/    //
+//                          /  \__$$ |                                            //
+//                          $$    $$/                                             //
+//                           $$$$$$/                                              //
+//                                                                                //
+//                            By: K1ngmar and rvan-mee                            //
+// ****************************************************************************** //
+
+#include <play-man/ROM/Rom.hpp>
+#include <play-man/utility/UtilFunc.hpp>
+#include <filesystem>
+#include <fstream>
+#include <iostream>
+#include <cstring>
+#include <assert.h>
+
+namespace GameBoy {
+
+    void RomHeader::ParseRawData(const std::vector<int8_t>& data)
+    {
+        assert(data.size() >= 0x14E);
+        title.fill('\0');
+        manufacturerCode.fill('\0');
+
+        // Nintendo Logo
+        for (uint32_t i = 0; i < nintendoLogoSize; i++)
+            nintendoLogo[i] = data[i + nintendoLogoIndex];
+
+        // Cartridge Title
+        for (uint32_t i = 0; i < romTitleSize; i++)
+        {
+            if (!(data[i + titleIndex] >= 32 && data[i + titleIndex] <= 126))
```

Comment:
Could [std::isprint()](https://en.cppreference.com/w/cpp/string/byte/isprint) be used for this? Also add a comment to why we stop reading the title when we encounter a non printable character.